src/pages/rooms/room-creation/room-creation.page.jsx

```diff
@@ -1,9 +1,180 @@
 import "./room-creation.page.scss";
+import { Endpoints, States } from "../../../constants";
+import { useState, useEffect } from "react";
+import {
+  Box,
+  TextField,
+  Button,
+  FormControl,
+  FormLabel,
+  RadioGroup,
+  FormControlLabel,
+  Radio,
+  InputLabel,
+  Select,
+  MenuItem,
+} from "@mui/material";
+import SendIcon from "@mui/icons-material/Send";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { useNavigate } from "react-router-dom";
+import AuthenticationService from "../../../services/authentication.service";
+import * as signalR from "@microsoft/signalr";
 
 const RoomCreationPage = () => {
-    return (
-        <div className="qcg-room-creation-page"></div>
-    );
+  const authenticationService = new AuthenticationService();
+
+  const [roomName, setRoomName] = useState();
+  const [category, setCategory] = useState(" ");
+
+  const [categories, setCategories] = useState([]);
+
+  const [isPublic, setIsPublic] = useState(true);
+  const [password, setPassword] = useState();
+
+  const navigationService = useNavigate();
+
+  const hubConnection = new signalR.HubConnectionBuilder()
+    .withUrl(Endpoints.SignalRRoomsListEndpointPrefix)
+    .configureLogging(signalR.LogLevel.Information)
+    .build();
+
+  const label = { inputProps: { "aria-label": "Checkbox demo" } };
+
+  const handleChange = (e) => {
+    setCategory(e.target.value);
+    console.log(e);
+  };
+
+  const getAllCategories = () => {
+    authenticationService
+      .get(Endpoints.Category)
+      .then((result) => {
+        setCategories(result.data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  useEffect(() => {
+    hubConnection.start().then(() => {
```

Comment:
This has nothing to do with signalR, getAllCategories should be outside of the hubConnection. hub connections should be completely removed from this page.

Reply:
Incase that, I should delete the all useEffect?